Source/Scene/ScreenSpaceCameraController.js

```diff
@@ -289,8 +293,33 @@ function ScreenSpaceCameraController(scene) {
   this._minimumRotateRate = 1.0 / 5000.0;
   this._minimumZoomRate = 20.0;
   this._maximumZoomRate = 5906376272000.0; // distance from the Sun to Pluto in meters.
+  this._minimumUndergroundPickDistance = 2000.0;
+  this._maximumUndergroundPickDistance = 10000.0;
+
+  /**
+   * The height of an invisible surface that the camera uses for underground navigation.
+   * For best results the height should be set just below the height of underground entities in the scene.
+   * Note that the camera can still zoom below this height if {@link minimumZoomDistance} is greater than 0.0.
```

Comment:
If `minimumZoomDistance` is 0.0 then it will approach a surface but never zoom through it. The controller treats collision detection and zoom distance as separate concepts, e.g. you can rotate the camera through the globe regardless of minimum zoom distance.